smithy-model/src/main/java/software/amazon/smithy/model/validation/NodeValidationVisitor.java

```diff
@@ -104,10 +104,21 @@ public enum Feature {
          * Emit a warning when a range trait is incompatible with a default value of 0.
          *
          * <p>This was a common pattern in Smithy 1.0 and earlier. It implies that the value is effectively
-         * required. However, chaning the type of the value by un-boxing it or adjusting the range trait would
-         * be a lossy tranformation when migrating a model from 1.0 to 2.0.
+         * required. However, changing the type of the value by un-boxing it or adjusting the range trait would
+         * be a lossy transformation when migrating a model from 1.0 to 2.0.
          */
-        RANGE_TRAIT_ZERO_VALUE_WARNING
+        RANGE_TRAIT_ZERO_VALUE_WARNING,
+
+        // Lowers severity of constraint trait validations to WARNING.
+        DISABLE_CONSTRAINTS;
```

Comment:
This name seems weird because it doesn't disable validation of constraints but lowers the severity. DISABLE_CONSTRAINTS -> IGNORE_CONSTRAINTS?

Reply:
Updated to `allowConstraintError` in [e5bcfd3](https://github.com/smithy-lang/smithy/pull/1949/commits/e5bcfd3a19ef1b1de341c87160f6ea5bcf41eab0)